src/EventListener/Api/PublishableEventListener.php

```diff
@@ -35,96 +36,167 @@ final class PublishableEventListener
     public function __construct(PublishableHelper $publishableHelper, ManagerRegistry $registry)
     {
         $this->publishableHelper = $publishableHelper;
+        // not unused, used by the trait
         $this->registry = $registry;
     }
 
-    public function __invoke(ViewEvent $event): void
+    private function checkMergeDraftIntoPublished(Request $request, object $data, bool $flushDatabase = false): object
     {
-        $request = $event->getRequest();
-        $data = $request->attributes->get('data');
-        if (!$this->publishableHelper->isPublishable($data)) {
-            return;
-        }
-
         $configuration = $this->publishableHelper->getConfiguration($data);
         $classMetadata = $this->getClassMetadata($data);
 
-        if ($request->isMethod(Request::METHOD_POST)) {
-            $this->handlePOSTRequest($classMetadata, $configuration, $data);
-        }
+        $publishedResourceAssociation = $classMetadata->getFieldValue($data, $configuration->associationName);
+        $draftResourceAssociation = $classMetadata->getFieldValue($data, $configuration->reverseAssociationName);
+
+        // the request is for a resource with an active publish date
+        // either a draft, if so it may be a published version we need to replace with
+        // or a published resource which may have a draft that has an active publish date
+        if (
+            $this->publishableHelper->isActivePublishedAt($data) &&
+            (
+                $publishedResourceAssociation ||
+                ($draftResourceAssociation && $this->publishableHelper->isActivePublishedAt($draftResourceAssociation))
+            )
+        ) {
+            $entityManager = $this->getEntityManager($data);
+            $meta = $entityManager->getClassMetadata(\get_class($data));
+            $identifierFieldName = $meta->getSingleIdentifierFieldName();
```

Comment:
```php
/** @var ClassMetadataInfo $meta */
$meta = $entityManager->getClassMetadata(\get_class($data));

Reply:
Maybe also add a check to ensure `$entityManager` is not null, otherwise the data is not an entity

Reply:
`getEntityManager` would throw an exception instead of being null. I think rightly so because publishable resources should be entities - otherwise none of this would work I don't think.